server/src/main/java/io/druid/server/namespace/cache/LookupCoordinatorManager.java

```diff
@@ -0,0 +1,597 @@
+      final @Smile ObjectMapper smileMapper,
+      final JacksonConfigManager configManager
+  )
+  {
+    this.listenerDiscoverer = listenerDiscoverer;
+    this.configManager = configManager;
+    this.httpClient = httpClient;
+    this.smileMapper = smileMapper;
+    executorService = MoreExecutors.listeningDecorator(
+        Executors.newScheduledThreadPool(10, Execs.makeThreadFactory("LookupCoordinatorManager--%s"))
+    );
+  }
+
+  void deleteOnHost(final URL url)
+      throws ExecutionException, InterruptedException, IOException
+  {
+    final AtomicInteger returnCode = new AtomicInteger(0);
+    final AtomicReference<String> reasonString = new AtomicReference<>(null);
+    LOG.debug("Dropping %s", url);
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.DELETE, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE),
+        makeResponseHandler(returnCode, reasonString),
+        Duration.millis(1_000) // TODO: configurable
+    ).get()) {
+      // 404 is ok here, that means it was already deleted
+      if (!HttpStatusCodes.isSuccess(returnCode.get()) || HttpStatusCodes.STATUS_CODE_NOT_FOUND != returnCode.get()) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+          StreamUtils.copyAndClose(result, baos);
+        }
+        catch (IOException e2) {
+          LOG.warn(e2, "Error reading response");
+        }
+
+        throw new IOException(
+            String.format(
+                "Bad lookup delete request [%d] : [%s]  Response: [%s]",
+                returnCode.get(),
+                reasonString.get(),
+                StringUtils.fromUtf8(baos.toByteArray())
+            )
+        );
+      } else {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Status: %s reason: [%s]", returnCode.get(), reasonString.get());
+        }
+      }
+    }
+  }
+
+  void updateAllOnHost(final URL url, Map<String, Map<String, Object>> knownLookups)
+      throws IOException, InterruptedException, ExecutionException
+  {
+    final AtomicInteger returnCode = new AtomicInteger(0);
+    final AtomicReference<String> reasonString = new AtomicReference<>(null);
+    final byte[] bytes;
+    try {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Loading up %d lookups to %s", knownLookups.size(), url);
+      }
+      bytes = smileMapper.writeValueAsBytes(knownLookups);
+    }
+    catch (JsonProcessingException e) {
+      throw Throwables.propagate(e);
+    }
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.POST, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .addHeader(HttpHeaders.Names.CONTENT_TYPE, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .setContent(bytes),
+        makeResponseHandler(returnCode, reasonString),
+        Duration.millis(1_000) // TODO: configurable
```

Comment:
I'll put configs in here shortly, it sounds like there's no more questions or major points of architecture changes that would need to be made.